mimic/test/test_identity_oskscatalog.py

```diff
@@ -0,0 +1,688 @@
+            json_request(self, self.root, self.verb,
+                         self.uri))
+
+        self.assertEqual(response.code, 401)
+        self.assertEqual(json_body['unauthorized']['code'], 401)
+
+    def test_invalid_service_id(self):
+        """
+        GET must have a valid service id otherwise 404.
+        """
+        self.headers.update({
+            'serviceid': [b'some-id']
+        })
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri,
+                         headers=self.headers))
+
+        self.assertEqual(response.code, 404)
+        self.assertEqual(json_body['itemNotFound']['code'], 404)
+
+    def test_list_only_internal_apis_available(self):
+        """
+        GET will return an empty listing when there are no External API
+        endpoint templates.
+        """
+        self.core.add_api(make_example_internal_api(self))
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri,
+                         headers=self.headers))
+
+        self.assertEqual(response.code, 200)
+        self.assertEqual(len(json_body['OS-KSCATALOG']), 0)
+        self.assertEqual(
+            len(json_body['OS-KSCATALOG:endpointsTemplates_links']),
+            0)
+
+    def test_list_single_template(self):
+        """
+        GET will return a endpoint template when there are External API
+        endpoint templates.
+        """
+        self.core.add_api(self.eeapi)
+
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri,
+                         headers=self.headers))
+
+        self.assertEqual(response.code, 200)
+        self.assertEqual(len(json_body['OS-KSCATALOG']), 1)
+        self.assertEqual(
+            len(json_body['OS-KSCATALOG:endpointsTemplates_links']),
+            0)
+
+    def test_list_single_template_external_and_internal_apis(self):
+        """
+        GET will only return the External API endpoint templates when they
+        are available,
+        even if there are also Internal APIs.
+        """
+        self.core.add_api(self.eeapi)
+        self.core.add_api(make_example_internal_api(self))
+
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri,
+                         headers=self.headers))
+
+        self.assertEqual(response.code, 200)
+        self.assertEqual(len(json_body['OS-KSCATALOG']), 1)
+        self.assertEqual(
+            len(json_body['OS-KSCATALOG:endpointsTemplates_links']),
+            0)
+
+    def test_multiple_external_apis(self):
+        """
+        GET can retrieve numerous External APIs that have External API Templates.
+        """
+        api_list = [self.eeapi]
+        for _ in range(10):
```

Comment:
Another place where a list comprehension would be a better fit, I think.  `api_list = [self.eeapi] + [ ... ]`